src/Common/Infrastructure/BackendResponse.cs

```diff
@@ -12,5 +15,48 @@ public BackendResponse(string content, string contentType, bool cameFromCache)
             ContentType = contentType;
             CameFromCache = cameFromCache;
         }
+
+        public void UpdateNextPageUrlWithQueryParameters(NonPagedQueryCollection requestQuery, string nextPageUrlOption)
+        {
+            if (string.IsNullOrWhiteSpace(nextPageUrlOption) || requestQuery.IsEmpty)
+                return;
+
+            const string numberPatternPlaceholder = "__NUMBER_PATTERN_PLACEHOLDER_THAT_WONT_BE_REGEX_ESCAPED__";
+            var nextPageUrlOptionWithPlaceholders = string.Format(nextPageUrlOption, numberPatternPlaceholder, numberPatternPlaceholder);
+            var nextPageUrlValePattern = Regex
+                .Escape(nextPageUrlOptionWithPlaceholders)
+                .Replace(numberPatternPlaceholder, "\\d+");
+
+            var parameters = requestQuery
+                .Aggregate("", (result, filterQueryParameter) => $"{result}&{filterQueryParameter.Key}={filterQueryParameter.Value}");
+
+            var isXml = ContentType.ToLower().Contains("xml");
+            var replaceData = new {
+                Pattern = isXml
+                    ? GetXmlPatternFor(nextPageUrlValePattern)
+                    : $"(\"volgende\"\\s*:\\s*\"{nextPageUrlValePattern})(\")",
+                Replacement = isXml
+                    ? $"$1{Xml.Escape(parameters)}$2"
+                    : $"$1{parameters}$2"
+            };
+
+            Content = Regex.Replace(
+                Content,
+                replaceData.Pattern,
+                replaceData.Replacement,
+                RegexOptions.IgnoreCase);
+        }
+
+        private string GetXmlPatternFor(string nextPageUrlValePattern)
+        {
+            return Regex.IsMatch(Content, "<feed xmlns=\"http://www.w3.org/2005/Atom\">")
```

Comment:
Is Regex IsMatch faster than string Contains?

Reply:
nope, forgot to replace after changing the patteren to a regular string

Reply:
for reference sake later on: https://theburningmonk.com/2012/05/performance-test-string-contains-vs-string-indexof-vs-regex-ismatch/ :)

Reply:
Does not differentiate between a lazy/non-lazy regex though ;-P
Anyway, replacing the Regex.IsMatch